### PR TITLE
Generate macpool range

### PR DESCRIPTION
**Pull request by SchSeba**
generate a range using a prefix of {0x02,<random>,<random>} if the range was not provided by the administrator.

---
This change is [Reviewable](https://reviewable.io/reviews/kubevirt/cluster-network-addons-operator/50)

### CODE REVIEW

**Comment by ovirt-infra** [comment]
All tests passed

File: pkg/network/kubemacpool.go [hunk 1]
```diff
@@ -61,3 +80,15 @@ func renderKubeMacPool(conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir str
...
 }
+
+func generateRandomMacPrefix() ([]byte, error) {
+	suffix := make([]byte, 2)
+	_, err := rand.Read(suffix)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	prefix := append([]byte{0x02}, suffix...)
```
Comment (phoracek) [inline pkg/network/kubemacpool.go]: This should be documented. Describe why do you use this `0x02` prefix

File: pkg/network/kubemacpool.go [hunk 2]
```diff
@@ -47,6 +53,19 @@ func renderKubeMacPool(conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir str
 		return nil, nil
 	}
 
+	if conf.KubeMacPool.StartPoolRange == "" || conf.KubeMacPool.EndPoolRange == "" {
```
Comment (phoracek) [inline pkg/network/kubemacpool.go]: please add an inline comment describing what happens here, mention how big the range is and the prefix for local unicast (IIUIC)

File: deploy/crds/network-addons-config-example.cr.yaml [hunk 3]
```diff
@@ -5,9 +5,7 @@ metadata:
   name: cluster
```
Comment (phoracek) [inline deploy/crds/network-addons-config-example.cr.yaml]: This file is generated. Please edit `pkg/components/components.go` `GetCR()`. Then run `make manifests`

Comment (SchSeba) [reply deploy/crds/network-addons-config-example.cr.yaml]: done

File: pkg/network/kubemacpool.go (hunk 2 above)
Comment (SchSeba) [reply pkg/network/kubemacpool.go]: done

File: pkg/network/kubemacpool.go (hunk 1 above)
Comment (SchSeba) [reply pkg/network/kubemacpool.go]: done add to the readme file and also a comment

**Comment by ovirt-infra** [comment]
All tests passed

File: README.md [hunk 4]
```diff
@@ -131,7 +129,10 @@ The operator allows administrator to deploy the [Kubemacpool](https://github.com
 This project allow to allocate mac addresses from a pool to secondary interfaces using
 [Network Plumbing Working Group de-facto standard](https://github.com/K8sNetworkPlumbingWG/multi-net-spec).
 
-Administrator need to specify a requested range
+*note:* Administrator can specify a requested range, if the range is not requested a random range will be provided
```
Comment (phoracek) [inline README.md]: please add **.** at the end of the sentence

File: README.md [hunk 5]
```diff
@@ -131,7 +129,10 @@ The operator allows administrator to deploy the [Kubemacpool](https://github.com
...
+
+The random range use a prefix of 0x02 that is a unicast local range.
+Two random hexadecimal numbers and three from 0x00 to 0xFF numbers
```
Comment (phoracek) [inline README.md]: I don't get this sentence, not here, nor in the code

**Comment by phoracek** [comment]
LGTM

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
Some tests failed

**Pull request merged**